app/src/processing/app/Base.java

```diff
@@ -1804,10 +1804,9 @@ static public LibraryList getLibraries() {
   }
 
   public File getDefaultSketchbookFolderOrPromptForIt() {
-
     File sketchbookFolder = BaseNoGui.getDefaultSketchbookFolder();
 
-    if (sketchbookFolder == null) {
+    if (sketchbookFolder == null && !isCommandLine()) {
```

Comment:
Does this not break below, when `sketchbookFolder` is `null` and `sketchbookFolder.exists()` is called? This seems to be a bit of a corner case, since it only occurs when the *default* sketchbook path (as returned by `Platform.getDefaultSketchbookFolder()`) is null. I'm not exactly sure how if this should not normally happen, or if this will happen if the default path does not exist (but judging from the `mkdirs()` a bit further down, it's probably the former).

Looking through the code to see how this works, it seems that the responsibility for resolving the sketchbook path might be too much spread out right now, so perhaps some additional refactoring should be done after this PR in this area.